src/util/XojMsgBox.cpp

```diff
@@ -102,23 +102,22 @@ void XojMsgBox::askQuestionWithMarkup(GtkWindow* win, std::string_view maintext,
 }
 
 void XojMsgBox::showErrorAndQuit(std::string& msg, int exitCode) {
-    /*
-     * This should be used for fatal errors, typically in early GUI startup (missing UI main file or so).
-     *
-     * Todo(gtk4): Figure out how to use a non-blocking dialog that would survive past the `exit` call.
-     * Putting the `exit()` in a callback is no good, as normal execution would continue in the background, most likely
-     * leading to a SegFault: that would again kill the dialog.
-     */
     GtkWidget* dialog = gtk_message_dialog_new_with_markup(defaultWindow, GTK_DIALOG_MODAL, GTK_MESSAGE_ERROR,
                                                            GTK_BUTTONS_OK, nullptr);
 
-    auto formattedMsg = xoj::util::OwnedCString::assumeOwnership(g_markup_escape_text(msg.c_str(), -1));
-    gtk_message_dialog_set_markup(GTK_MESSAGE_DIALOG(dialog), formattedMsg.get());
-    if (defaultWindow != nullptr) {
-        gtk_window_set_transient_for(GTK_WINDOW(dialog), defaultWindow);
+    gtk_message_dialog_set_markup(GTK_MESSAGE_DIALOG(dialog), msg.c_str());
+
+    // We use a hack to ensure the app does not exit until the user has read the error message and closed the popup
+    bool done = false;
+
+    xoj::popup::PopupWindowWrapper<XojMsgBox> popup(GTK_DIALOG(dialog), [&done](int) { done = true; });
+
+    popup.show(defaultWindow);
+
+    while (!done) {
+        // Let the main loop run so the popup pops up and can be interacted with
+        g_main_context_iteration(g_main_context_default(), true);
```

Comment:
You don't. For now, I would std::ignore it.
Even, it would be good, to have this in general, this was only an idea, to ensure correct handling through the application, while you refactor all code using Dialogs. After you are done, search replace all occurrences with void.

When you decide to keep it, in order to prevent others, doing it wrong, you should either  
- std::ignore all cases, where we can't return, 
- refactor all the code, where you can't return: 
  - when a constructor blocks the return, its object must be split into a generator and a constructor without the problematic code.